src/main/java/org/broadinstitute/hellbender/tools/walkers/ReferenceConfidenceVariantContextMerger.java

```diff
@@ -545,14 +545,18 @@ private GenotypesContext mergeRefConfidenceGenotypes(final VariantContext vc,
                             ? calculators.getInstance(ploidy, maximumAlleleCount).genotypeIndexMap(perSampleIndexesOfRelevantAlleles, calculators) //probably horribly slow
                             : genotypeIndexMapsByPloidy[ploidy];
                     final int[] PLs = generatePL(g, genotypeIndexMapByPloidy);
-                    final int[] AD = g.hasAD() ? AlleleSubsettingUtils.generateAD(g.getAD(), perSampleIndexesOfRelevantAlleles) : null;
-                    genotypeBuilder.PL(PLs).AD(AD);
-                //clean up low confidence hom refs for better annotations later
+                    genotypeBuilder.PL(PLs);
+                // clean up low confidence hom refs for better annotations later
                 } else if (GenotypeGVCFsEngine.excludeFromAnnotations(g)) {
                     genotypeBuilder.alleles(Collections.nCopies(ploidy, Allele.NO_CALL));
                 }
+                if (g.hasAD()) {
```

Comment:
I would move this up instead so the code is less branched/duplicative… and you've also lost the else-if case when we have ADs.  like

```
if (g.hasPL() || g.hasAD()) {
  # do common stuff (like perSampleIndexesOfRelevantAlleles)
  if (g.hasPL() {
     # do PL only stuff
   }
   if (g.hasAD()) {   
      # do AD stuff
   }
} else if (GenotypeGVCFsEngine.excludeFromAnnotations(g)) {
   # as is
}

Reply:
Do we want the GenotypeGVCFsEngine.excludeFromAnnotations(g) if there is no AD? I didn't think so, which I why I set it up the way I did